session_rpcserver.go

```diff
@@ -810,13 +810,21 @@ func (s *sessionRpcServer) ListAutopilotFeatures(ctx context.Context,
 			return nil, err
 		}
 
+		// We check whether we should upgrade because of unknown privacy
+		// flags.
+		_, err = session.Deserialize(f.PrivacyFlags)
+		if errors.Is(err, session.ErrUnknownPrivacyFlag) {
+			upgrade = true
+		}
```

Comment:
should we return the error otherwise? if we cant parse it here, it will error later on anyways. 
So either should:
1) return the error here or
2) log it and make upgrade = true  so that user doesnt go and try to register to this feature since it will fail anyways right?

Reply:
I returned the error :+1: